### PR TITLE
Move legacy node version tests to GitHub actions

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

This PR moves legacy node versions (6.x, 8.x) tests to GitHub versions. The new legacy version test workflow will build Babel on latest node but run unit tests on legacy node. By doing so it paves the way to bump Babel _build_ requirements to `>= 10.13`. Babel 7 still supports node 6, but to build babel project, we requires higher node versions.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/11952"></a>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          make -j test-ci-coverage
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+
+  legacy-node-version-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
```
Comment (fisker) [inline .github/workflows/ci.yml:26]: Did you mean `false`? `true` is default

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Use Node.js latest
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: "*"
```
Comment (fisker) [inline .github/workflows/ci.yml:14]: `node-version` is not required, can be removed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JLHwung) [reply .github/workflows/ci.yml:26]: Oh I meant for failing fast. Didn't know this is default. Thanks.

Comment (fisker) [reply .github/workflows/ci.yml]: Why failing fast?

File: babel.config.js [hunk 3]
```diff
@@ -109,7 +109,7 @@ module.exports = function (api) {
         "@babel/proposal-object-rest-spread",
         { useBuiltIns: true, loose: true },
       ],
-      compileDynamicImport ? dynamicImportUrlToPath : null,
+      dynamicImportUrlToPath,
       compileDynamicImport ? "@babel/plugin-proposal-dynamic-import" : null,
```
Comment (nicolo-ribaudo) [inline babel.config.js:113]: Why is this change necessary?

Comment (JLHwung) [reply babel.config.js]: Ah I see. `import(url.pathToFileURL())` is meant to be published. I have added a new `test-legacy` BABEL_ENV to `babel.config.js`.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JLHwung) [reply .github/workflows/ci.yml]: They are strategy for legacy node version tests. I think they are highly correlated, if one of them is failed the other one will probably fail, too(Regardless of network issues or the change indeed will fail on node 8).

**Comment by nicolo-ribaudo** [comment]
Awesome!

File: .travis.yml [hunk 4]
```diff
@@ -41,10 +41,9 @@ matrix:
         directories:
           - $HOME/AppData/Local/Temp/chocolatey
     # Continue node_js matrix
-    - node_js: "6"
     - node_js: "12"
     - node_js: "10"
-    - node_js: "8"
+    # Node.js 6 and 8 are tested on GitHub versions
```
Comment (nicolo-ribaudo) [inline .travis.yml:46]:
```suggestion
    # Node.js 6 and 8 are tested on GitHub actions
```
